index.ts

```diff
@@ -578,3 +596,5 @@ declare module '@hapi/joi' {
      */
     extractOne<T>;
 }
+
+export default JoiExtractType;
```

Comment:
Just wondering did you check if it is possible to export only the type-extraction function instead?
Instead of using
`JoiExtractType.extractType<schema>`
We could just
`JoiExtractType<schema>`

Reply:
It is possible, but I had to export all types and interfaces outside of the JoiExtractType module while keeping the function interfaces within the module. This now leads to being able to import any type / interface and use it independently. `extractType` is now the default export. for example:

```ts
import * as Joi from 'joi';
import extractType, { extendsGuard } from './index';

const extJoi = Joi.extend({} as Joi.Extension);

// Unknown types or AnySchema defaults to type any
const any_schema = Joi.any();
type extractAny = extractType<typeof any_schema>;
export const extractedAny: extractAny = 'anything';
```

Please let me know if this is sufficient, or if you'd like me to rollback to the earlier implementation.

And also, big thanks to you for this library. It is very helpful!